rpc/examples/echo/tests/echo.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { test, expect } from "@playwright/test";
+
+test("receives responses", async ({ page }) => {
+  await page.goto("/");
+  const table: [string, string[]][] = [
+    ["unary-wrtc", ["hello"]],
+    ["multi-wrtc", ["h", "e", "l", "l", "o", "?"]],
+    ["bidi-wrtc", ["o", "n", "e", "t", "w", "o"]],
+    ["unary-direct", ["hello"]],
+    ["multi-direct", ["h", "e", "l", "l", "o", "?"]],
+    // gRPC-web does not yet support bidirectional streaming so we expect to
+    // only receive a response to our first request.
+    ["bidi-direct", ["o", "n", "e"]],
+  ];
+
+  for (const [testID, expected] of table) {
+    await expect(page.getByTestId(testID).getByTestId("message")).toHaveCount(
+      expected.length,
+    );
+    const messages = page.getByTestId(testID).getByTestId("message");
+    const contents = await messages.allTextContents();
+
+    expect(contents).toStrictEqual(expected);
```

Comment:
Locaters in Playwright are lazy and stateless. They're a description of an element search, and they don't evaluate to an actual element until used inside an `expect` call (or if you use a method like `await locator.allTextContents()`).

Accordingly, it makes sense to assign locators only once and re-use them. In general, I also find splitting locator creation away from calling `expect` makes Playwright tests much easier to read.

Finally, this `toHaveCount` check is redundant with `expect(contents).toStrictEqual(expected);` below. If count of elements is off, then `contents` will fail the call to `toStrictEqual`, because it will have different length

```suggestion
    const messages = page.getByTestId(testID).getByTestId("message");
    const contents = await messages.allTextContents();

    expect(contents).toStrictEqual(expected);
```

Reply:
Hm, the suggested implementation is flaky, at least for Firefox. It seems like `allTextContents()` might validly return nothing if we reach the `expect` call before the elements are loaded.

Reply:
Oh oops, my bad, splitting out `allTextContents` from the `expect` will cause flakiness. You're right, we should check the count. We can then lean on `toContainText` to check contents.

This works for me locally:

```suggestion
    const messages = page.getByTestId(testID).getByTestId("message");

    await expect(messages).toHaveCount(expected.length)
    await expect(messages).toContainText(expected);
```

Reply:
sweet, confirmed that this is stable now